bindings/rust/s2n-tls-tokio/examples/client.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use clap::Parser;
+use s2n_tls::raw::{config::Config, error::Error, security::DEFAULT_TLS13};
+use s2n_tls_tokio::TlsConnector;
+use std::fs;
+use tokio::net::TcpStream;
+
+const DEFAULT_CERT: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/examples/certs/cert.pem");
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[clap(short, long, default_value_t = String::from(DEFAULT_CERT))]
+    trust: String,
+    addr: String,
+}
+
+async fn run_client(trust_pem: &[u8], addr: &String) -> Result<(), Error> {
+    let mut config = Config::builder();
+    config.set_security_policy(&DEFAULT_TLS13)?;
+    config.trust_pem(trust_pem)?;
+    let client = TlsConnector::new(config.build()?);
+
+    let stream = TcpStream::connect(addr).await.expect("Failed to connect");
+    client.connect("localhost", stream).await?;
+    // TODO: echo
+    
+    Ok(())
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
```

Comment:
maybe just return `Result<(), Box<dyn, std::error::Error>` like the [s2n-quic examples](https://github.com/aws/s2n-quic/blob/a67ea08c375b81f91ea6e7c3ef3154b98d1ccb4b/examples/echo/src/bin/quic_echo_client.rs#L14) so you can use the ? operator for all errors instead of `expect`